src/main/java/org/truffleruby/language/methods/SharedMethodInfo.java

```diff
@@ -61,6 +63,11 @@ public SharedMethodInfo(
         this.parseName = parseName;
         this.notes = notes;
         this.argumentDescriptors = argumentDescriptors;
+
+        if (backtraceName.equals("dig")) {
+            // Why isn't isInternal set?
```

Comment:
Ruby core method are no internal anymore since a while.
Internal is really meant as implementation details, not as public core library methods.
Java core is still internal, but there are plans to change that as well.

Reply:
So what should be `isInternal`? Is anyone using the API anymore? We can find another way to check if it's core library if needed but funny that this API seems to say it's the right thing but unused.

Reply:
The JavaDoc says:
> Returns <code>true</code> if this root node should be considered internal and not be shown to
> a guest language programmer. This method has effect on tools and guest language stack traces.

And basically we do want & need to show core library methods in tools and strack traces.
Java core methods use the file:line of the caller, so they need special handling, we'll add something like RootNode#countsTowardsStackTraceLimit() to handle that.

Potentially some `Truffle::` methods could be considered internal, but I think it would just hurt debugging and not worth the effort. `isInternal` also affects what is shown if an exception escapes the embedder `Context`.

Reply:
This property has a different definition everywhere you look!

https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/source/Source.SourceBuilder.html#internal-boolean-

> Internal sources are those that aren't created by user, but rather inherently present by the language system.